src/components/checkbox/styled-components.js

```diff
@@ -127,7 +130,43 @@ export const Root = styled('label', props => {
   };
 });
 
-export const Checkmark = styled('span', props => {
+const getToggleCheckMarkStyles = props => {
+  const {$checked, $theme, $disabled} = props;
+  const {animation, colors} = $theme;
+  const toggleSVG =
+    `<svg width="32" height="32" viewBox="0 0 32 32" fill="none" xmlns="http://www.w3.org/2000/svg"><g filter="url(#filter0_d)"><rect x="4" y="3" width="24" height="24" rx="4" fill="` +
+    ($disabled ? colors.mono500 : 'white') +
+    `"/><rect width="2" height="8"  x="15" y="9" rx="1" fill="` +
+    getBackgroundColor(props) +
+    `"/></g><defs><filter id="filter0_d" x="0" y="0" width="32" height="32" filterUnits="userSpaceOnUse" color-interpolation-filters="sRGB"><feFlood flood-opacity="0" result="BackgroundImageFix"/><feColorMatrix in="SourceAlpha" type="matrix" values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"/><feOffset dy="1"/><feGaussianBlur stdDeviation="2"/><feColorMatrix type="matrix" values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.32 0"/><feBlend mode="normal" in2="BackgroundImageFix" result="effect1_dropShadow"/><feBlend mode="normal" in="SourceGraphic" in2="effect1_dropShadow" result="shape"/></filter></defs></svg>`;
+  return {
+    position: 'relative',
+    flex: '0 0 auto',
+    borderRadius: $theme.borders.useRoundedCorners
+      ? $theme.borders.radius200
+      : null,
+    display: 'inline-block',
+    backgroundColor: getBackgroundColor(props),
+    width: '40px',
+    height: '16px',
+    marginTop: '4px',
+    marginBottom: '4px',
+    marginLeft: '4px',
+    marginRight: '4px',
+    ':after': {
+      position: 'absolute',
+      top: '-8px',
+      transitionProperty: 'margin-left',
+      transitionDuration: animation.timing700,
+      transitionTimingFunction: animation.easeOutCurve,
+      marginLeft: $checked ? '16px' : '-4px',
+      content: `url('data:image/svg+xml;utf8,` + toggleSVG + `')`,
+      color: 'black',
+    },
+  };
+};
```

Comment:
It looks like the notch on the toggle handle is a little bit off-center:
![image](https://user-images.githubusercontent.com/875591/44747528-430b6e00-aac2-11e8-9c9b-aed4ebe5e853.png)

Should be perfectly centered based on mocks:
![image](https://user-images.githubusercontent.com/875591/44747578-5cacb580-aac2-11e8-9523-8e43efe2202a.png)

Reply:
Text also looks like it's too close to the toggle element (when it's off). @nguyensomniac how many pixels of margin should we have here?

Reply:
Also, looks like shadow is still showing when control is disabled:
![image](https://user-images.githubusercontent.com/875591/44747666-8f56ae00-aac2-11e8-93d3-b4383605caf2.png)

Mocks have the shadow removed here:
![image](https://user-images.githubusercontent.com/875591/44747684-9aa9d980-aac2-11e8-8afa-1e93357b9c70.png)

Reply:
Hover style in this diff seems to just change opacity:
![image](https://user-images.githubusercontent.com/875591/44747735-bb722f00-aac2-11e8-9860-f4639435a473.png)

I think its supposed to make the track and notch a darker blue color?
![image](https://user-images.githubusercontent.com/875591/44747761-c88f1e00-aac2-11e8-9e1d-fc3322b3f7b3.png)

Reply:
@nguyensomniac please, check the mocks. Seems like they are not consistent. The color of dash is not consistent with color of toggle itself. Shadow is not everywhere

Reply:
@schnerd fixed obvious issues with styles

Reply:
Nice, looks centered correctly now.

>Seems like they are not consistent. The color of dash is not consistent with color of toggle itself. Shadow is not everywhere

Maybe those are on purpose? In my opinion the disabled toggle looks better without shadow–pretty sure that's intentional.

Also, should things be aligned left in the flexbox?
![image](https://user-images.githubusercontent.com/875591/44761730-1d4d8b80-aaf9-11e8-8a99-c20af0cc8960.png)

After we resolve those I'm happy to stamp.

Reply:
In my experience usually label and toggle are left and right aligned in container given. I wait for @nguyensomniac to check this assumption

Reply:
Shadows shouldn't change. I'll investigate

Reply:
Fyi, the color of the thumb becomes `primary400` when hovered/active, as in different from the color of the background

Reply:
What is thumb? The small dash in center?
The small dash has no change in color (or almost no change) even if the slider color changes to gray. Is it correct?

Reply:
Yes, the dash in the center

Reply:
fixed